src/lib/components/DatePickerPanel.js

```diff
@@ -36,6 +36,12 @@ export default class DatePickerPanel extends React.Component {
     const { currentValue, confirmDisabled } = this.state;
     const disabledTimeFns = this.disabledTime();
 
+    const uiLocale = [browser.i18n.getUILanguage().replace('_', '-'), 'en-US'];
+    const dtf = new Intl.DateTimeFormat(uiLocale, {hour: 'numeric'});
+    const timeFormat = (dtf.resolvedOptions().hour12)
+      ? 'h:mm a'
+      : 'HH:MM';
+
```

Comment:
I wonder if we could/should pull this into the new `time-formats.js` file in a future patch?